backend/repository/thrift_compile.go

```diff
@@ -174,19 +174,232 @@ type ResultSpec struct {
+	return CompileThriftCode(b, thriftAbsPath, thriftRootPath)
+}
+
+// CompileThriftCode compiles thrift code into a structured module.
+func CompileThriftCode(rawCode []byte, thriftAbsPath string, thriftRootPath string) (*Module, error) {
+	thriftFile := relPath(thriftRootPath, thriftAbsPath)
+	thriftFileName := strings.Split(filepath.Base(thriftFile), ".")[0]
+
+	program, err := idl.Parse(rawCode)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse thrift program")
 	}
-	return ConvertModule(compiledModule, program, thriftRootPath), nil
+
+	var namespaces []*Namespace
+	includedModules := make(map[string]*IncludedModule)
+
+	for _, header := range program.Headers {
+		switch header := header.(type) {
+		case *ast.Include:
+			targetPath := filepath.Join(filepath.Dir(thriftAbsPath), header.Path)
+			incModuleDetails, err := CompileThriftFile(targetPath, thriftRootPath)
+			if err == nil {
+				includedModules[incModuleDetails.Name] = &IncludedModule{
+					Name:   incModuleDetails.Name,
+					Module: incModuleDetails,
+					Line:   header.Line,
+				}
+			}
+		case *ast.Namespace:
+			namespaces = append(namespaces,
+				&Namespace{
+					Name:  header.Name,
+					Scope: header.Scope,
+					Line:  header.Line,
+				})
+		}
+	}
+
+	constants := make(map[string]*Constant)
+	types := make(map[string]*TypeSpec)
+	services := make(map[string]*ServiceSpec)
+
+	for _, definition := range program.Definitions {
+		switch definition := definition.(type) {
+		case *ast.Constant:
+			constants[definition.Name] = &Constant{
+				Name: definition.Name,
+				File: thriftFile,
+				Line: definition.Line,
+				Type: &TypeSpec{
+					Name: definition.Type.String(),
+				},
+				Value: constantAstValue(definition.Value),
+			}
+		case *ast.Typedef:
+			types[definition.Name] = &TypeSpec{
+				Name: definition.Name,
+				File: thriftFile,
+				Line: definition.Line,
+				TypeDefTarget: &TypeSpec{
+					Name: definition.Type.String(),
+				},
+				Annotations: convertAstAnnotations(definition.Annotations),
+			}
+		case *ast.Enum:
+			types[definition.Name] = &TypeSpec{
+				Name: definition.Name,
+				File: thriftFile,
+				Line: definition.Line,
+				EnumType: &EnumSpec{
+					Name:  definition.Name,
+					File:  thriftFile,
+					Items: convertAstEnumItems(definition.Items),
+				},
+			}
+		case *ast.Struct:
+			types[definition.Name] = &TypeSpec{
+				Name: definition.Name,
+				File: thriftFile,
+				Line: definition.Line,
+				StructType: &StructTypeSpec{
+					Kind:   convertStructType(definition.Type),
+					Fields: convertAstFields(definition.Fields),
+				},
+				Annotations: convertAstAnnotations(definition.Annotations),
+			}
+		case *ast.Service:
+			services[definition.Name] = &ServiceSpec{
+				Name:        definition.Name,
+				File:        thriftFile,
+				Line:        definition.Line,
+				Functions:   convertAstFunctions(definition.Functions),
+				Annotations: convertAstAnnotations(definition.Annotations),
+			}
+		}
```

Comment:
do we want to add a default case where we store a `&TypeSpec` in `types`  with `{ Name, File, Line, Annotations }` ?